client/js/components/user/CustomerSettings/CustomerSettingsContact.vue

```diff
@@ -0,0 +1,241 @@
+<template>
+  <div class="layout u-pl">
+    <dp-editable-list
+      ref="editableList"
+      :entries="contacts"
+      :translation-keys="translationKeys"
+      @reset="resetForm"
+      @saveEntry="id => dpValidateAction('contactData', () => createOrUpdateContact(id), false)">
+      <template v-slot:list="contact">
+        <p
+          class="weight--bold u-mt"
+          v-text="contact.attributes.title" />
+        <span
+          class="block"
+          v-text="contact.attributes.phoneNumber" />
+        <span
+          class="block"
+          v-text="contact.attributes.eMailAddress" />
+        <span
+          class="block"
+          v-html="contact.attributes.text" />
+        <span v-text="Translator.trans('customer.contact.visibleText', {isVisible: contact.attributes.visible})" />
+      </template>
+      <template v-slot:form>
+        <div
+          id="contactForm"
+          data-dp-validate="contactData">
+          <dp-input
+            id="contactTitle"
+            v-model="customerContact.title"
+            class="u-mb-0_75"
+            data-cy="contactTitle"
+            data-dp-validate-error="error.customer.contact.title"
+            :placeholder="Translator.trans('customer.contact.title')"
+            required
+            type="text" />
+          <dp-input
+            id="phoneNumber"
+            v-model="customerContact.phoneNumber"
+            autocomplete="tel"
+            class="u-mb-0_75"
+            data-cy="phoneNumber"
+            data-dp-validate-error="error.customer.contact.phone_or_email"
+            pattern="^(\+?)(-| |[0-9]|\(|\))*$"
+            :placeholder="Translator.trans('customer.contact.phone_number')"
+            :required="phoneIsRequired"
+            type="tel"
+            @input="input => setRequiredEmail(input)" />
```

Comment:
why is this necessary when the phoneNumber is set with v-model?

Reply:
It sets the state if email should be a required field or not. Not the value.

Reply:
CustomerContact as moved to data, so we can use it here and don't need this extra method ([b953896](https://github.com/demos-europe/demosplan-core/pull/2139/commits/b953896dc33c223b3f01c83319023bdae9f26e28)).